docs/components/contacts.rst

```diff
@@ -4,8 +4,170 @@ Contacts
+              'firstname' => 'Bob',
+              //...
+          );
+
+          // Optionally check for identifier fields to determine if the lead is unique
+          $uniqueLeadFields    = $this->fieldModel->getUniqueIdentiferFields();
+          $uniqueLeadFieldData = array();
+
+          // Check if unique identifier fields are included
+          $inList = array_intersect_key($leadFields, $uniqueLeadFields);
+          foreach ($inList as $k => $v) {
+              if (empty($query[$k])) {
+                  unset($inList[$k]);
+              }
+
+              if (array_key_exists($k, $uniqueLeadFields)) {
+                  $uniqueLeadFieldData[$k] = $v;
+              }
+          }
+
+          // If there are unique identifier fields, check for existing leads based on lead data
+          if (count($inList) && count($uniqueLeadFieldData)) {
+              $existingLeads = $this->leadRepository->getLeadsByUniqueFields(
+                  $uniqueLeadFieldData,
+                  $leadId // If a currently tracked lead, ignore this ID when searching for duplicates
+              );
+              if (!empty($existingLeads)) {
+                  // Existing found so merge the two leads
+                  $lead = $this->leadModel->mergeLeads($lead, $existingLeads[0]);
+              }
+
+              // Get the lead's currently associated IPs
+              $leadIpAddresses = $lead->getIpAddresses();
+
+              // If the IP is not already associated, do so (the addIpAddress will automatically handle ignoring
+              // the IP if it is set to be ignored in the Configuration)
+              if (!$leadIpAddresses->contains($ipAddress)) {
+                  $lead->addIpAddress($ipAddress);
+              }
+          }
+
+          // Set the lead's data
+          $this->leadModel->setFieldValues($lead, $leadFields);
+
+          // Save the entity
+          $this->leadModel->saveEntity($lead);
+      }
+  }
+
+Contact tracking
+****************
+
+Contacts get tracked by two cookies. The first cookie registers the ID of the Contact that's tracked by Mautic.
+The second is to track the Contact's activity for the current session. This defaults to 30 minutes and resets during each Contact interaction.
+  
+``mautic_session_id`` holds the value of the Contact's current session ID.  That value is then name of the cookie that holds the Contact's ID. 
+
+Review the sample code on how to obtain the currently tracked Contact.
+
+.. note:: As of Mautic 2.2.0, a cookie is also placed on any domain with mtc.js embedded. Ensure that Mautic's CORS settings allow the domain. This contains the ID of the currently tracked Contact.
+
+.. code-block:: PHP
+
+  <?php
+  // plugins/HelloWorldBundle/Services/ContactTrackingService.php
+
+  declare(strict_types=1);
+
+  namespace MauticPlugin\HelloWorldBundle\Services;
+
+  use Mautic\LeadBundle\Entity\Lead;
+  use Mautic\LeadBundle\Tracker\ContactTracker;
+
+  class ContactTrackingService
+  {
```

Comment:
Also refactored this code sample. [Old docs here](https://developer.mautic.org/#manipulating-contacts-(leads)).